service/mantle/shopify/ApiServices.xml

```diff
@@ -458,17 +458,36 @@
         </in-parameters>
         <out-parameters>
             <parameter name="collectData"/>
+            <parameter name="error"/>
         </out-parameters>
         <actions>
             <entity-find-one entity-name="mantle.product.category.ProductCategory" value-field="productCategory">
                 <field-map field-name="productCategoryId" from="productCategoryId"/>
             </entity-find-one>
+            
+            <if condition="!productCategory">
+                <set field="error" value="true"/>
+                <message type="warning">The product category ${productCategoryId} does not exist</message>
+                <return/>
+            </if>
+            <if condition="productCategory.pseudoId == productCategory.productCategoryId">
+                <set field="error" value="true"/>
+                <message type="warning">The product category ${productCategoryId} does not have shopifyId</message>
+                <return/>
+            </if>
 
             <entity-find-one entity-name="mantle.product.ProductIdentification" value-field="productIdentification">
                 <field-map field-name="productId" from="productId"/>
                 <field-map field-name="productIdTypeEnumId" value="PidtShopify"/>
             </entity-find-one>
-            <set field="collectData" value="."/>
+
+            <if condition="!productIdentification">
+                <set field="error" value="true"/>
+                <message type="warning">The product ${productId} does not have a value in ProductIdentification entity</message>

```

Comment:
Message: The product ${productId} does not have a Shopify Id on the in ProductIdentification entity

Reply:
Done

Reply:
I avoided the "in" preposition